src/plugins/maps_legacy/public/map/service_settings.js

```diff
@@ -205,12 +227,17 @@ export class ServiceSettings {
   }
 
   async getFileLayerFromConfig(fileLayerConfig) {
-    const fileLayers = await this._emsClient.getFileLayers();
-    return fileLayers.find((fileLayer) => {
-      const hasIdByName = fileLayer.hasId(fileLayerConfig.name); //legacy
-      const hasIdById = fileLayer.hasId(fileLayerConfig.id);
-      return hasIdByName || hasIdById;
-    });
+    let fileLayers = [];
+    try {
+      fileLayers = await this._emsClient.getFileLayers();
+      return fileLayers.find((fileLayer) => {
+        const hasIdByName = fileLayer.hasId(fileLayerConfig.name); //legacy
+        const hasIdById = fileLayer.hasId(fileLayerConfig.id);
+        return hasIdByName || hasIdById;
+      });
+    } catch (err) {
+      return null;
```

Comment:
Do we properly handle the null return here? Can you add tests for this?

Reply:
This PR fixes custom WMS load bug(maps tiles layer), in region maps feature, it will request maps tiles layer and files layer from maps service. WMS allow user to custom maps tiles layer. The file layer config methods `getEMSHotLink` and `loadFileLayerConfig` will need to return null because it can't fetch any file layer data when can't access to our maps service(no Internet), if not return null, it return still block WMS load. @Shivamdhar works on offline custom geojson(maps files layer), so that in no internet case, users still can use region maps.

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/42f877cf9424f347f114b9c25c8b290291be6904/src/plugins/maps_legacy/public/map/service_settings.js#L243-L252

yes, we have already have a test for it. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/42f877cf9424f347f114b9c25c8b290291be6904/src/plugins/maps_legacy/public/map/service_settings.test.js#L397